lib/coordinates.rb

```diff
@@ -0,0 +1,59 @@
+class Coordinates
+  def initialize(coords)
+    @coords = coords.split(" ")
+  end
+
+  def are_diagonal?
+    not are_vertical? and not are_horizontal?
+  end
+
+  def are_vertical?
+    expected_length = @coords.length
+    y_coords = @coords.map do |coord|
+      coord[1..-1].to_i
+    end
+    y_coords.uniq.length == 1
+  end
+
+  def are_horizontal?
+    expected_length = @coords.length
+    x_coords = @coords.map do |coord|
+      coord[0]
+    end
+    x_coords.uniq.length == 1
+  end
+
+  def intersects?(coords)
+    coords_a = coords.to_a
+    for index in (0..@coords.length) do
```

Comment:
Is the range from 0..3? Or should it be 0..2? For an actual array length of 3. If so on the second question, then would it be ...`(0..(@coords.length - 1))`?

Reply:
In this case, the range is going to make the for-loop go from 0 to 3 and not 0 to 2 since the indexing would be 0, 1, and 2 for that array. But further down the test `coords_a.include?(coord)` will ignore the nil that would be produced by accessing the array at an index that doesn't exist.
Very good point though.